testing/e2e/repos_loader.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Title        repos_loader.sh
+# Description  Load all mocked git repositories making their "local remote_url" available via env vars
+#=====================================================================================================
+
+source ./testing/e2e/logging.sh
+
+REPOS_DIR=""
+REPO_SMALL_DIR=""
+
+function prepare_test_repositories() {
+  REPOS_DIR=${TEST_TMPDIR}/repos
+  if [ -e "${REPOS_DIR}" ]; then
+    rm -rf ${REPOS_DIR}
+  fi
+
+  if [[ -d ${repos_dir} ]]; then
+    log_info "Repositories folder is already initialized, skipping"
+  fi
+
+  mkdir -p $REPOS_DIR
+
+  cp ./testing/testdata/small-repo.tar.gz $REPOS_DIR
+
+  cwd=$(pwd)
+  cd $REPOS_DIR
+  tar zxf small-repo.tar.gz
```

Comment:
I still vote for constructing the repo on the fly rather than tar.gz...

Reply:
Creating repos on the fly in addition to the concerns I've raised before would force the framework to support dynamic commit hashes. 
IMO we need to add more test use cases and evolve the framework accordingly avoiding premature optimizations, we're talking about `14Kb`..